x-pack/plugins/cases/public/components/all_cases/use_all_cases_state.tsx

```diff
@@ -80,6 +80,47 @@ const validateQueryParams = (queryParams: QueryParams): QueryParams => {
   return { ...queryParams, perPage, sortOrder };
 };
 
+/**
+ * Parses filter options from the URL.
+ * If a filter option is defined as an array in DEFAULT_FILTER_OPTIONS, this function
+ * ensures it's returned as an array even if URLSearchParams provides it as a single value.
+ * URLSearchParams.getAll() returns an empty array if the param is not defined in the URL.
+ * URLSearchParams.get() returns null if the param is not defined in the URL and in case of
+ * multiple values, it returns the first value only
+ */
+const parseURLWithFilterOptions = (search: string) => {
+  const urlParams = new URLSearchParams(search);
+
+  const paramKeysWithTypeArray = Object.entries(DEFAULT_FILTER_OPTIONS)
+    .map(([key, val]) => (Array.isArray(val) ? key : undefined))
+    .filter(Boolean);
+
+  const parsedUrlParams: { [key in string]: string[] | string | null } = {};
+  urlParams.forEach((_, key) => {
+    if (paramKeysWithTypeArray.includes(key)) {
```

Comment:
I didn't knew that! I will go for a slightly different approach using `for (const [key, value] of urlParams.entries()) {` as it allows me to get rid of getAll/get and make it easier to read

Reply:
Ah, but my point was rather that `urlParams.forEach` iterates over key-value pairs and not just `keys`.

(This is just a nit 😅) Since we are iterating over key-value pairs we will be calling `urlParams.getAll(key)` and running the same code for the same key when we have something like `status=A&status=B&status=C`.

Maybe better to do `urlParams.keys().forEach`?

Reply:
Yeah, I understood your point. It made me realise that I don't need to use getAll/get, and I believe it made the code more straightforward to read (or at least I hope so). Given the number of keys we're dealing with, I'm not too concerned about any minor performance issues that might arise from a key being processed multiple times. In fact, depending on what getAll does internally, it might even be faster to handle it the way I'm proposing, wdyt?

Reply:
> as it allows me to get rid of getAll/get

I missed this part in the previous comment 😀 

>  In fact, depending on what getAll does internally, it might even be faster to handle it the way I'm proposing, wdyt?

Yup, all good 👍